server/src/api/gitLab/gitLabIntegration.ts

```diff
@@ -0,0 +1,169 @@
+  IntegrationConfig,
+  InvalidTokenError,
+} from '../integration';
+
+const authorizePath = '/oauth/authorize';
+const tokenPath = '/oauth/token';
+const redirectBaseURL = `${process.env.FRONTEND_BASE_URL}/oauth/redirect`;
+
+const oauth2 = (key: string, secret: string, host: string) =>
+  new OAuth2(key, secret, host, authorizePath, tokenPath);
+
+export const GitLabIntegration: IntegrationEntry = {
+  defaultConfig: { host: 'gitlab.com' },
+  configOptions: [
+    { field: 'projectId' },
+    { field: 'consumerkey' },
+    { field: 'privatekey', secret: true },
+  ],
+  getIntegrationProvider: (config, tokens) =>
+    new GitLabImporter(config, tokens),
+  getOAuthProvider: (config) => ({
+    instance: config.host,
+    authorizationURL: (roadmapId) =>
+      new Promise((resolve, reject) => {
+        const state = randomBytes(10).toString('hex');
+
+        const redirectUri = encodeURIComponent(
+          `${redirectBaseURL}?roadmapId=${roadmapId}&integrationName=gitLab`,
+        );
+        resolve({
+          url: `https://${config.host}${authorizePath}?client_id=${config.consumerkey}&response_type=code&state=${state}&scope=read_api&redirect_uri=${redirectUri}`,
+          token: '',
+          tokenSecret: '',
+        });
+      }),
+    swapOAuthToken: ({ verifierToken, token, tokenSecret, roadmapId }) =>
+      new Promise((resolve, reject) => {
+        const redirectUri = `${redirectBaseURL}?roadmapId=${roadmapId}&integrationName=gitLab`;
+
+        oauth2(
+          config.consumerkey,
+          config.privatekey,
+          `https://${config.host}`,
+        ).getOAuthAccessToken(
+          verifierToken,
+          {
+            grant_type: 'authorization_code',
+            redirect_uri: redirectUri,
+          },
+          (err, access_token, refresh_token, result) => {
+            if (err) reject(err);
+            else
+              resolve({
+                accessToken: access_token,
+                refreshToken: refresh_token,
+              });
+          },
+        );
+      }),
+  }),
+};
+
+class GitLabImporter implements IntegrationProvider {
+  private readonly projectId: string;
+  private readonly oauth2: OAuth2;
+  private readonly accessToken: string;
+  private lastUsedRedirectUri: string;
+
+  constructor(
+    { projectId, consumerkey, privatekey, host }: IntegrationConfig,
+    tokens: { accessToken: string },
+  ) {
+    this.projectId = projectId || '';
+    this.lastUsedRedirectUri = '';
+    this.accessToken = tokens.accessToken;
+    this.oauth2 = oauth2(consumerkey, privatekey, host);
+  }
+
+  setLastUsedRedirectUri(uri: string) {
+    this.lastUsedRedirectUri = uri;
+  }
+  getLastUsedRedirectUri() {
+    return this.lastUsedRedirectUri;
+  }
+
+  async boards() {
+    const response = await this.fetch<{ id: string; name: string }[]>('boards');
+    return (response as any[]).map((board) => ({
+      id: board.id as string,
+      name: board.name as string,
+    }));
```

Comment:
```suggestion
    const response = await this.fetch<{ id: number; name: string }[]>('boards');
    return response.map((board) => ({
      id: `${board.id}`,
      name: board.name,
    }));
```

I suppose this is the root cause for the `toString()`s. Here practically `any` is coerced to `{ id: string; name: string }[]`, and I suppose the `id` isn't actually a `string`.

So the correct place to convert would be the boundary from `any` to a known type.


Same might happen with `columns()` and others, if there are more type mismatches.